internal/app/machined/internal/phase/api/acpi.go

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package api
+
+import (
+	"log"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+func sync() {
+	syncdone := make(chan struct{})
+
+	go func() {
+		defer close(syncdone)
+		unix.Sync()
+	}()
+
+	log.Printf("waiting for sync...")
+
+	for i := 29; i >= 0; i-- {
+		select {
+		case <-syncdone:
+			log.Printf("sync done")
+			return
+		case <-time.After(time.Second):
+		}
+		if i != 0 {
+			log.Printf("waiting %d more seconds for sync to finish", i)
+		}
+	}
+
+	log.Printf("sync hasn't completed in time, aborting...")
```

Comment:
i know it's not part of the actual pr itself, but we probably want to drop the `, aborting...`. makes it unclear if the shutdown is being aborted or the sync.